framework/base/Profiler.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\base;
+
+use Yii;
+use yii\log\Logger;
+
+/**
+ * Profiler is the class for application profiling.
+ *
+ * Profiler is used as a core component of Yii application.
+ *
+ * @see [[Yii::beginProfile()]]
+ * @see [[Yii::endProfile()]]
+ *
+ * @author cronfy <cronfy@gmail.com>
+ * @author Qiang Xue <qiang.xue@gmail.com>
+ * @since 2.0.12
+ */
+class Profiler extends Component
+{
+
+    /**
+     * Marks the beginning of a code block for profiling.
+     * This has to be matched with a call to [[end()]] with the same category name.
+     * The begin- and end- calls must also be properly nested.
+     *
+     * @param string $token token for the code block
+     * @param string $category the category of this profiler message (used for logging)
+     * @see end()
+     */
+    public function begin($token, $category = 'application')
```

Comment:
Why unused? `$context` should be logged. And this is public method of public component, people may always use `Yii::$app->logger->begin('my-code', 'application', $context)`. Not mention that changing that in Yii 2.1 will create unnecessary BC break. 

And personally I don't really see much sense of releasing this in current state. You're creating API, that can't be changed easily in the future. We should decide how this should work in first place - this is not so urgent issue, there is no need for rush and temporary solution.